presto-native-execution/README.md

```diff
@@ -96,17 +85,90 @@ checks.  When we need one it can be added.
 ```
+* It is recommended to use some sort of IDE on Mac. CLion is an option.
 
 ### Setting up on Linux (CentOs 8 or later)
 
-See `scripts/setup-linux.sh`
+* Clone the whole Presto repository.
+* Run `presto/presto-native-execution/scripts/setup-linux.sh`
+
+### Setting up local Presto environment on MacOs (with optional IntelliJ and CLion) 
+Clone the whole Presto repository.
+
+Close IntelliJ and CLion
+
+In local Presto repo
+* `git fetch upstream`
+* `git co upstream/master`
+* `mvn clean install -DskipTests -T1C -pl -presto-docs`
+
+Run IntelliJ
+* Edit/Create `HiveQueryRunnerExternal` Run/Debug Configuration (alter paths accordingly).
+  * Main class: `com.facebook.presto.hive.HiveExternalWorkerQueryRunner`.
+  * VM options: `-ea -Xmx5G -XX:+ExitOnOutOfMemoryError -Duser.timezone=America/Bahia_Banderas -Dhive.security=legacy`.
+  * Working directory: `$MODULE_DIR$`
+  * Environment variables: `PRESTO_SERVER=/Users/spershin/git/presto/presto-native-execution/cmake-build-debug/presto_cpp/main/presto_server;DATA_DIR=/Users/spershin/Desktop/data;WORKER_COUNT=0`
+  * Use classpath of module: choose `presto-native-execution` module.
+* Edit create `TestHiveQueriesUsingJSON` Test Run/Debug Configuration (alter paths accordingly).
+  * Class: `com.facebook.presto.nativeworker.TestHiveQueriesUsingJSON`
+  * VM Options: `-ea -DPRESTO_SERVER=/Users/spershin/git/presto_cpp/cmake-build-debug/presto_cpp/main/presto_server -DDATA_DIR=/Users/spershin/Desktop/data`
+  * Working directory: `$MODULE_WORKING_DIR$`
+
+In "presto/presto-native-execution" run:
+* `make -C presto-native-execution submodules`
+* `git/presto/presto-native-execution % sudo chown -R $(whoami) /usr/local/bin /usr/local/lib /usr/local/sbin; sudo chown -R $(whoami) /usr/local/bin /usr/local/lib /usr/local/bin; sudo chown -R $(whoami) /usr/local/bin /usr/local/lib /usr/local/lib; chmod u+w /usr/local/bin /usr/local/lib /usr/local/sbin; chmod u+w /usr/local/bin /usr/local/lib /usr/local/bin; chmod u+w /usr/local/bin /usr/local/lib /usr/local/lib; ./scripts/setup-macos.sh`
+  
+Run CLion:
+* File->Close Project if any is open.
+* Open `presto/presto-native-execution` folder as CMake project and wait till CLion loads/generates make files, symbols, etc.
+* Edit configuration for `presto_server` module (alter paths accordingly).
+  * Program arguments: `--logtostderr=1 --v=1 --etc_dir=/Users/spershin/git/presto/presto-native-execution/etc`
+  * Working directory: `/Users/spershin/git/presto/presto-native-execution`
+* Edit menu CLion->Preferences->Build, Execution, Deployment->CMake
+  * CMake options: `-DVELOX_BUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Debug`
+  * Build options: `-- -j 12`
+* Edit menu CLion->Preferences->Editor->Code Style->C/C++
+  * Scheme: `Project`
+* To enable clang format you need
+  * Open any h or cpp file in the editor and select `Enable ClangFormat` by clicking `4 spaces` rectangle in the status bar (bottom right) which is next to `UTF-8` bar.
```

Comment:
Would it be helpful to add a screenshot here?

Reply:
We can do that. :)

Reply:
Thank you for adding a screenshot. I'm surprised that .clang-format doesn't specify the number of spaces and we need to specify it separately.